### PR TITLE
Enable scneario management for different configs

### PR DESCRIPTION
**Pull request by FabianHofmann**
From the release notes:

The software now supports running the workflow with different settings within the same directory. A new config section ``run`` was created that specifies under which scenario ``name`` the created resources, networks and results should be stored. If ``name`` is not specified, the workflow uses the default paths. The entry ``dedicated_cutout`` specifies whether the run should use cutouts from the default root directory and use run-specific cutouts.

@fneum @p-glaum  any suggestion on how to rename `dedicated_cutouts`? No so happy with this key...

Note this is a non-breaking change.

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [x] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes.

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -123,37 +128,37 @@ rule build_shapes:
...
         ch_popgdp='data/bundle/je-e-21.03.02.xls'
     output:
-        country_shapes='resources/country_shapes.geojson',
-        offshore_shapes='resources/offshore_shapes.geojson',
-        europe_shape='resources/europe_shape.geojson',
-        nuts3_shapes='resources/nuts3_shapes.geojson'
-    log: "logs/build_shapes.log"
+        country_shapes="resources/" + RDIR + "country_shapes.geojson",
+        offshore_shapes="resources/" + RDIR + "offshore_shapes.geojson",
+        europe_shape="resources/" + RDIR + "europe_shape.geojson",
```
[the remark this answers is not in the document]
Comment (FabianHofmann) [reply Snakefile:133]: The shapes are dependent on the selected countries so I maid it dependent on the run. You are right the `europe_shape` is redundant here, but I'd keep it this way for a consistent data flow.

File: Snakefile [hunk 2]
```diff
@@ -123,37 +128,37 @@ rule build_shapes:
...
+        nuts3_shapes="resources/" + RDIR + "nuts3_shapes.geojson"
```
[the remark this answers is not in the document]
Comment (p-glaum) [reply Snakefile:134]: here I meant only to comment on the nuts3_shapes

File: Snakefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (p-glaum) [reply Snakefile:133]: ah okay. here I only meant to comment on the Europe shape. But for consistency it makes sense

**Comment by FabianHofmann** [comment]
@fneum any objection to merge this?

File: Snakefile [hunk 3]
```diff
@@ -13,9 +13,14 @@ if not exists("config.yaml"):
 
 configfile: "config.yaml"
 
-COSTS="resources/costs.csv"
+run = config.get("run", {})
+RDIR = run.get("name", "") + "/" if run.get("name", "") else ""
```
Comment (fneum) [inline Snakefile:17]: shorter alternative?
```suggestion
RDIR = run["name"] + "/" if run.get("name") else ""
```

File: doc/configuration.rst [hunk 4]
```diff
@@ -28,13 +28,24 @@ Top-level configuration
...
-``scenario``
-============
+``run``
+=======
 
 It is common conduct to analyse energy system optimisation models for **multiple scenarios** for a variety of reasons,
 e.g. assessing their sensitivity towards changing the temporal and/or geographical resolution or investigating how
 investment changes as more ambitious greenhouse-gas emission reduction targets are applied.
 
+The ``run`` section is used for running and storing scenarios with different configurations which are not covered by :ref:`wildcards`. It determines the path at which resources, networks and results are stored. Therefore the user can run different configurations within the same directory. If a run with a non empty name should use the default cutouts, set ``dedicated_cutouts`` to `false`.
```
Comment (fneum) [inline doc/configuration.rst:38]:
```suggestion
The ``run`` section is used for running and storing scenarios with different configurations which are not covered by :ref:`wildcards`. It determines the path at which resources, networks and results are stored. Therefore the user can run different configurations within the same directory. If a run with a non-empty name should use cutouts shared across runs, set ``shared_cutouts`` to `true`.       
```

**Review by fneum — CHANGES_REQUESTED** [review]
Nice! Especially, since it keeps paths as they were if no run-name is specified!

Just a few minor comments:

Could you document new `run` options also at

https://github.com/PyPSA/pypsa-eur/blob/master/doc/configtables/toplevel.csv

File: Snakefile (hunk 3 above)
Comment (FabianHofmann) [reply Snakefile:17]: good one :)

File: config.default.yaml [hunk 5]
```diff
@@ -9,6 +9,11 @@ logging:
   level: INFO
   format: '%(levelname)s:%(name)s:%(message)s'
 
+run: 
+  name: "" # use this to keep track of runs with different settings
+  dedicated_cutouts: True # set to false to use the default cutout(s)
```
[the remark this answers is not in the document]
Comment (FabianHofmann) [reply config.default.yaml:14]: yes, that's a good idea

**Pull request merged**